### ISSUE TITLE
Option to have a system BEEP when a match is found

### ISSUE THREAD

**Issue by blandyuk** [hashcat/hashcat#3643]
It would be great if hashcat could do a system BEEP when a match is found. Default would be OFF of course. This would be VERY helpful as I would know something has been cracked without having to check [s]tatus every so often. Plus, us crackers would know instantly, which is handy if time is an issue.

I have something similar with an automated console app which monitors the hashcat output file, and BEEPs when something is found.

Maybe use a simple command parameter like: --beep

There is an issue as this works for a few hashes, but if there are many 1000's of hashes, might be worth delaying the next BEEP for 5-10 seconds? Maybe have another parameter to control time delay between each? --beep-delay=5

I would use this feature 100%. I do runs on my laptop while working, so BEEPing would alert me to something being found.

**Comment by roycewilliams** [comment]
Duplicate of #2319 and IIRC maybe a couple of others :D Still worth exploring IMO - feels like there could be a way to make it work "well enough" for the common use cases. Current workaround is to tail+grep the potfile, etc.

**Comment by Chick3nman** [comment]
The issues brought up in https://github.com/hashcat/hashcat/issues/2319 are probably not going to be so simple to work around if we BEEP on hashes cracked. But we could potentially BEEP on hashcat "status" change. Perhaps when we go from Running -> Cracked, we attempt to fire a bell/beep/attention/etc. That way we dont lag terminals trying to fire it for 5000 cracked hashes per second but we still offer additional alerting to someone who may be waiting for a single important time sensitive hash.

**Comment by blandyuk** [comment]
It would not BEEP for every one, just once, then delay doing it again for 5-10 secs. Won't cause an issue, but give us crackers an audio notification of something being cracked. If more are cracked in that 5-10 secs, it BEEPs again, then waits.

**Comment by davidbolvansky** [comment]
What about to handle it generally with plugin system? hashcat would load plugin.so file and would run code from this plugin during some events..

**Comment by Chick3nman** [comment]
An extension/plugin system for that sort of thing would need to be built and I worry about the added complexity and upkeep it may require, especially for something that can be achieved by a wrapper and is mostly a convenience feature.

**Comment by davidbolvansky** [comment]
But in the end we may end up with tens of “cool” small features which would be much more invasive in codebase then properly designed extension mechanism via plugins.

Custom event handler seems like a reasonable thing to have.

**Comment by Chick3nman** [comment]
True, there's potential for quite a few interesting features if we expose parts of the API through a custom plugin loader.
